sources/api/static-pods/src/main.rs

```diff
@@ -0,0 +1,338 @@
+/*!
+# Background
+
+static-pods ensures static pods are running as defined in settings.
+
+It queries for all existing static pod settings, then configures the system as follows:
+* If the pod is enabled, it creates the manifest file in the pod manifest path that kubelet is
+  configured to read from and populates the file with the base64-decoded manifest setting value.
+* If the pod is enabled and the manifest file already exists, it overwrites the existing manifest
+  file with the base64-decoded manifest setting value.
+* If the pod is disabled, it ensures the manifest file is removed from the pod manifest path.
+*/
+
+#![deny(rust_2018_idioms)]
+
+#[macro_use]
+extern crate log;
+
+use simplelog::{Config as LogConfig, LevelFilter, SimpleLogger};
+use snafu::{ensure, OptionExt, ResultExt};
+use std::collections::HashMap;
+use std::env;
+use std::fs;
+use std::path::{Path, PathBuf};
+use std::process;
+use std::str::FromStr;
+
+use model::modeled_types::Identifier;
+
+// FIXME Get from configuration in the future
+const DEFAULT_API_SOCKET: &str = "/run/api.sock";
+const API_SETTINGS_URI: &str = "/settings";
+
+const STATIC_POD_DIR: &str = "/etc/kubernetes/static-pods";
+
+type Result<T> = std::result::Result<T, error::Error>;
+
+/// Query the API for the currently defined static pods
+async fn get_static_pods<P>(socket_path: P) -> Result<Option<HashMap<Identifier, model::StaticPod>>>
+where
+    P: AsRef<Path>,
+{
+    debug!("Querying the API for settings");
+
+    let method = "GET";
+    let uri = API_SETTINGS_URI;
+    let (code, response_body) = apiclient::raw_request(&socket_path, uri, method, None)
+        .await
+        .context(error::APIRequest { method, uri })?;
+    ensure!(
+        code.is_success(),
+        error::APIResponse {
+            method,
+            uri,
+            code,
+            response_body,
+        }
+    );
```

Comment:
Do you actually need this? Both [`apiclient::raw_request`](https://github.com/bottlerocket-os/bottlerocket/blob/8ea251400615cd275d830b690a445414451516d0/sources/api/apiclient/src/lib.rs#L85) and `schnauzer::get_settings` (through [`schnauzer::get_json`](https://github.com/bottlerocket-os/bottlerocket/blob/8ea251400615cd275d830b690a445414451516d0/sources/api/schnauzer/src/lib.rs#L59:14)) have this check already